pkg/registry/rbac/escalation_check.go

```diff
@@ -44,6 +46,56 @@ func EscalationAllowed(ctx context.Context) bool {
 	return false
 }
 
+var roleResources = map[schema.GroupResource]bool{
+	rbac.SchemeGroupVersion.WithResource("clusterroles").GroupResource(): true,
+	rbac.SchemeGroupVersion.WithResource("roles").GroupResource():        true,
+}
+
+// RoleEscalationAuthorized checks if the user associated with the context is explicitly authorized to escalate the role resource associated with the context
+func RoleEscalationAuthorized(ctx context.Context, a authorizer.Authorizer) bool {
+	if a == nil {
+		return false
+	}
+
+	user, ok := genericapirequest.UserFrom(ctx)
+	if !ok {
+		return false
```

Comment:
Do these `return false` lines merit a log (at least an Info.V4)? If we can't pull a user/requestInfo off the context in this call, something is broken, right?